### PR TITLE
[WIP] MGMT-11563: Add ppc64le & s390x RHCOS ISOs to configmap

### PR DESCRIPTION
**Pull request by jacobemery**
Include ppc64le and s390x 4.11.0 RHCOS ISOs to Podman's configmap.yml to match the recently added corresponding multiarch release image for OCP, which includes these architectures. ARM and x86 were already included in the list of RHCOS images.

@mkowalski had a [PR](https://github.com/openshift/assisted-service/commit/76c665404295aef6f822f49479f8ec9e58ce7fd2) merged 8 days ago that added a 4.11.0 multiarch OCP image to Podman's configmap. This multiarch image included ppc64le and s390x, but these architectures were not added to the RHCOS ISOs list. This PR adds them.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [X] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @mkowalski 

## Checklist

- [X] Title and description added to both, commit and PR.
- [X] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [X] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: deploy/podman/configmap.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ data:
...
...ncies/rhcos/4.10/4.10.16/rhcos-4.10.16-x86_64-live.x86_64.iso","version":"410.84.202205191234-0"},{"openshift_version":"4.10","cpu_architecture":"arm64","url":"https://mirror.openshift.com/pub/openshift-v4/aarch64/dependencies/rhcos/4.10/4.10.16/rhcos-4.10.16-aarch64-live.aarch64.iso","version":"410.84.202205191023-0"},{"openshift_version":"4.11","cpu_architecture":"x86_64","url":"https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.11/4.11.0/rhcos-4.11.0-x86_64-live.x86_64.iso","version":"411.86.202207150124-0"},{"openshift_version":"4.11","cpu_architecture":"arm64","url":"https://mirror.openshift.com/pub/openshift-v4/aarch64/dependencies/rhcos/4.11/4.11.0/rhcos-4.11.0-aarch64-live.aarch64.iso","version":"411.86.202207130959-0"},{"openshift_version":"4.11","cpu_architecture":"s390x","url":"https://mirror.openshift.com/pub/openshift-v4/s390x/dependencies/rhcos/4.11/4.11.0/rhcos-4.11.0-s390x-live.s390x.iso","version":"411.86.202208190940-0"},{"openshift_version":"4.11","cpu_architecture":"ppc64le","url":"https://mirror.openshift.com/pub/openshift-v4/ppc64le/dependencies/rhcos/4.11/4.11.0/rhcos-4.11.0-ppc64le-live.ppc64le.iso","version":"411.86.202208190941-0"}]'
```
Comment (romfreiman) [inline deploy/podman/configmap.yml:24]: no way to have it as multiline to be more readable?

**Comment by romfreiman** [comment]
just verifying that @vrutkovs promotion script wont get crazy

**Comment by mkowalski** [comment]
/retitle MGMT-11563: Add ppc64le & s390x RHCOS ISOs to configmap

**Comment by vrutkovs** [comment]
You'd need to edit `data/default_os_images.json` and run `make generate-configuration`

ocp-bump would be okay (if not I'll fix it)

**Comment by jacobemery** [comment]
> You'd need to edit `data/default_os_images.json` and run `make generate-configuration`
> 
> ocp-bump would be okay (if not I'll fix it)

Added commit for data/default_os_images.json

**Comment by jacobemery** [comment]
> You'd need to edit `data/default_os_images.json` and run `make generate-configuration`
> 
> ocp-bump would be okay (if not I'll fix it)

Was able to run 'make generate-configuration' and will commit soon.

**Comment by jacobemery** [comment]
> You'd need to edit `data/default_os_images.json` and run `make generate-configuration`
> 
> ocp-bump would be okay (if not I'll fix it)
>
> Was able to run 'make generate-configuration' and will commit soon.

Committed, thanks for the help! 

Had to re-commit the last one because I hadn't setup my GitHub email so it appeared to come in from a different user.

**Comment by jacobemery** [comment]
What can I do to pass the rest of these tests?

File: config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml [hunk 2]
```diff
@@ -1,58 +1,37 @@
 apiVersion: agent-install.openshift.io/v1beta1
 kind: AgentServiceConfig
-metadata:
-  name: agent
+metadata: {name: agent}
```
Comment (vrutkovs) [inline config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml:3]: This is unnecessary and `verify-generated-code` doesn't like it

Comment (jacobemery) [reply config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml]: Reverted this change and the other seemingly unnecessary YAML reformatting changes that the 'make generate-configuration' command did, but kept the new P/Z osImages parameters. [Commit](https://github.com/openshift/assisted-service/pull/4333/commits/794af8173723ee8c6e12072b53cf7aa04445022d).

File: config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml [hunk 3]
```diff
@@ -56,3 +56,11 @@ spec:
     openshiftVersion: '4.11'
     url: https://mirror.openshift.com/pub/openshift-v4/aarch64/dependencies/rhcos/4.11/4.11.0/rhcos-4.11.0-aarch64-live.aarch64.iso
     version: 411.86.202207130959-0
+  - cpuArchitecture: ppc64le
+    openshiftVersion: '4.11', 
+    url: https://mirror.openshift.com/pub/openshift-v4/ppc64le/dependencies/rhcos/4.11/4.11.0/rhcos-4.11.0-ppc64le-live.ppc64le.iso
+    version: 411.86.202208190941-0
+  - cpuArchitecture: s390x
+    openshiftVersion: 4.11
```
Comment (mkowalski) [inline config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml:64]: Should be `'4.11'` (note the quotes)

Comment (jacobemery) [reply config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml]: Thanks! Fixed

File: deploy/podman/configmap.yml (hunk 1 above)
Comment (jacobemery) [reply deploy/podman/configmap.yml]: Not a bad idea now that there are even more, I'll leave that up to the ai team.

**Comment by mkowalski** [comment]
@jacobemery 

I have noticed

`"Failed to populate image store: failed to create minimal iso template for version map[cpu_architecture:ppc64le openshift_version:4.11 url:https://mirror.openshift.com/pub/openshift-v4/ppc64le/dependencies/rhcos/4.11/4.11.0/rhcos-4.11.0-ppc64le-live.ppc64le.iso version:411.86.202208190941-0]: no grub.cfg found, possible paths are [EFI/redhat/grub.cfg EFI/fedora/grub.cfg]\n"`

Can you make sure those ISOs are conform?

**Comment by jacobemery** [comment]
> @jacobemery
> 
> I have noticed
> 
> `"Failed to populate image store: failed to create minimal iso template for version map[cpu_architecture:ppc64le openshift_version:4.11 url:https://mirror.openshift.com/pub/openshift-v4/ppc64le/dependencies/rhcos/4.11/4.11.0/rhcos-4.11.0-ppc64le-live.ppc64le.iso version:411.86.202208190941-0]: no grub.cfg found, possible paths are [EFI/redhat/grub.cfg EFI/fedora/grub.cfg]\n"`
> 
> Can you make sure those ISOs are conform?

Looks like neither ppc64le and s390x ISOs' grub.cfg files are in the standard paths. 

ppc64le has a grub.cfg in boot/grub/grub.cfg. I can't find a grub.cfg in s390x folders.

This is probably a larger issue that we'll need to work through with other teams. I'll reach out to Rom to see where we should go from here.

**Comment by mkowalski** [comment]
/retitle [WIP] MGMT-11563: Add ppc64le & s390x RHCOS ISOs to configmap

**Comment by jacobemery** [comment]
Chatted with Nick Carboni who helped me add another path to search for grub.cfg for ppc64le, which got around the above error. 

s390x doesn't use grub and therefore doesn't have a grub.cfg file. All the components (kernel, initramfs, rootfs) are there to create a minimal ISO, but since it doesn't have a grub.cfg file these components would have to be pointed to in a more direct way. This would be an exception for s390x and would take some work from the broader team. For now, I just skipped the creation of a minimal ISO for s390x. A full ISO was successfully created by the assisted-image-service.

After that I encountered another error with finding the isolinux.cfg file, which neither architecture has. I nullified the errors for finding it and assisted-image-service was able to create a minimal ISO for ppc64le. This would be an exception that would have to be worked in for both ppc64le and s390x for a more formal solution to this. But the generation of the minimal ISO works without the isolinux.cfg file, so this shouldn't be too big an issue.

Now I'll test whether these ISOs can boot RHCOS nodes. If that works, I'll try a full test-run to install OCP on each architecture.

And if _that_ works, then we can return to this PR and start to work toward a more formal solution to the above errors.

**Comment by mkowalski** [comment]
> Now I'll test whether these ISOs can boot RHCOS nodes. If that works, I'll try a full test-run to install OCP on each architecture.
> 
> And if _that_ works, then we can return to this PR and start to work toward a more formal solution to the above errors.

Thanks a lot, the effort is really appreciated! I hope with this work you do here we will be able to enable Assisted as the “installer of choice” for all those platforms

**Comment by jacobemery** [comment]
Closing this PR as a [newer version](https://github.com/openshift/assisted-service/pull/4796) was merged 2 weeks ago.

**Pull request closed without merging**